### PR TITLE
PF-1212: Use deleteReferenceResourceForResourceType instead of deleteReferenceResource

### PR DESCRIPTION
**Pull request by yuhuyoyo**

### CODE REVIEW

**Review by ddietterich — APPROVED** [review]
LGTM - thanks!

File: service/src/main/java/bio/terra/workspace/service/resource/referenced/ReferencedResourceService.java [hunk 1]
```diff
@@ -117,8 +117,8 @@ public void deleteReferenceResource(
   }
 
   /**
-   * Delete a reference. The only state we hold for a reference is in the metadata database so we
```
Comment (zloery) [inline service/src/main/java/bio/terra/workspace/service/resource/referenced/ReferencedResourceService.java:120]: Are the `deleteReferenceResource` and `ResourceDao.deleteResource` still used anywhere else? if not, you could clean them up

**Review by zloery — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/service/resource/referenced/ReferencedResourceService.java (hunk 1 above)
Comment (yuhuyoyo) [reply service/src/main/java/bio/terra/workspace/service/resource/referenced/ReferencedResourceService.java:120]: it is still used by workspaceapicontroller#deleteDataReference. This API is [deprecated](https://github.com/DataBiosphere/terra-workspace-manager/blob/16e3dbb90e70d6c4aba606fda146c6456c6300c4/service/src/main/resources/api/service_openapi.yaml#L385) but i'm not sure if there's still any service calling this endpoint.

Comment (zloery) [reply service/src/main/java/bio/terra/workspace/service/resource/referenced/ReferencedResourceService.java:120]: ah, good point. I believe Rawls still calls those endpoints in CWB, so it's fine to leave that for now.

**Pull request merged**